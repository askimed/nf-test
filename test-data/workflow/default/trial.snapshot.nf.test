nextflow_workflow {

    name "Test workflow"
    script "test-data/workflow/default/trial.nf"
    workflow "trial"

    test("Should run without failures") {
        when {
            params {
                outdir = "tests/results"
            }
            workflow {
                """
                input[0] = Channel.of('a','b')
                """
            }
        }

        then {
            //check if test case succeeded
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }
}