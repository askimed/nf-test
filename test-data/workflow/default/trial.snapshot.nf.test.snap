{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "a",
                    "b"
                ],
                "1": [
                    "a.txt:md5,0cc175b9c0f1b6a831c399e269772661",
                    "b.txt:md5,92eb5ffee6ae2fec3ad71c777531578f"
                ],
                "lukas": [
                    "a",
                    "b"
                ],
                "sebastian": [
                    "a.txt:md5,0cc175b9c0f1b6a831c399e269772661",
                    "b.txt:md5,92eb5ffee6ae2fec3ad71c777531578f"
                ]
            }
        ],
        "timestamp": "2023-08-28T14:23:59+0000"
    }
}